Content.Server/Theta/Impostor/Systems/ImpostorMagicRevolverSystem.cs

```diff
@@ -0,0 +1,163 @@
+using System.Numerics;
+using Content.Server.Administration.Managers;
+using Content.Server.DoAfter;
+using Content.Server.Mind;
+using Content.Server.Popups;
+using Content.Server.Roles.Jobs;
+using Content.Server.Theta.Impostor.Components;
+using Content.Shared.Database;
+using Content.Shared.DoAfter;
+using Content.Shared.Examine;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Mind;
+using Content.Shared.Mind.Components;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Systems;
+using Content.Shared.Popups;
+using Content.Shared.Projectiles;
+using Content.Shared.Roles.Jobs;
+using Content.Shared.Theta.Impostor.Components;
+using Content.Shared.Weapons.Ranged.Components;
+using Content.Shared.Weapons.Ranged.Systems;
+using Content.Shared.Theta.Impostor.Events;
+using Robust.Shared.Map;
+using Robust.Shared.Network;
+
+namespace Content.Server.Theta.Impostor.Systems;
+
+public sealed class ImpostorMagicRevolverSystem : EntitySystem
+{
+    [Dependency] private PopupSystem _popupSys = default!;
+    [Dependency] private DoAfterSystem _doAfterSys = default!;
+    [Dependency] private SharedGunSystem _gunSys = default!;
+    [Dependency] private JobSystem _jobSys = default!;
+    [Dependency] private IBanManager _banMan = default!;
+    [Dependency] private MobStateSystem _mobSys = default!;
+    [Dependency] private MindSystem _mindSys = default!;
+
+    private const string MagicProjectileProtoId = "ImpostorMagicBullet";
+    private const string RegularProjectileProtoId = "BulletMagnum";
+    private const string CaptainJobId = "Captain";
+    private const string CommRelayProtId = "ImpostorCommRelay";
+    private const uint MagicBanDuration = 60*8; //minutes
+    
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ImpostorMagicBulletComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<ImpostorMagicBulletComponent, UseInHandEvent>(OnBulletUseInHand);
+        SubscribeLocalEvent<ImpostorMagicBulletComponent, ProjectileHitEvent>(OnBulletHit);
+        SubscribeLocalEvent<ImpostorMagicRevolverComponent, InteractUsingEvent>(OnRevolverUse, before: new[] { typeof(SharedGunSystem) });
```

Comment:
```suggestion
        SubscribeLocalEvent<ImpostorMagicBulletComponent, ProjectileHitEvent>(OnBulletHit);

        SubscribeLocalEvent<ImpostorMagicRevolverComponent, InteractUsingEvent>(OnRevolverUse, before: new[] { typeof(SharedGunSystem) });
```
просто для читабельности